### DISCUSSION TITLE
Is Thread.Priority ignored on Linux?

### DISCUSSION THREAD

**Discussion by packaman** [dotnet/runtime#59160] [Q&A] [answered]
I'm running a netcoreapp3.1\linux-arm build of my application on a fairly vanilla Debian platform. I want one thread running at higher priority than the others but setting Thread.Priority=Highest doesn't seem to be having any effect. I can see (using 'top -H') that all the native threads are running at normal priority and this is consistent with the behaviour of my application. 

I've tried running as root and setcap 'cap_sys_nice=+ep' to no effect.

Am I missing something or is this just not going to work?

Thanks,

Martin

**Comment by huoyaoyuan** [comment, marked as answer]
`Thread.Priority` is converted to posix priority here:

https://github.com/dotnet/runtime/blob/072cbae7b4fc5198476a94fa6673b4076c65e7cc/src/coreclr/pal/src/thread/thread.cpp#L1042-L1221

**Reply by packaman** [reply]
Thanks. So it looks like it should work but I can't get it to work for me. I've posted a follow-up on [Stack Overflow](https://stackoverflow.com/questions/69389768/how-do-i-get-thread-priority-to-work-on-linux-net-core-3-1)
